ci/release-qualification/pipeline.yml

```diff
@@ -27,14 +27,9 @@ steps:
         required: false
     if: build.source == "ui"
 
-  - id: build-x86_64
-    label: Build x86_64
-    command: bin/ci-builder run stable bin/pyactivate -m ci.test.build x86_64
-    timeout_in_minutes: 60
-    agents:
-      queue: builder-linux-x86_64
-
-  - wait: ~
+  # There is intentionally no build command here since we want to make sure we
+  # use a properly tagged version when testing the upcoming release, and don't
+  # build our own version just for the tests.
```

Comment:
I don't think this will work as is! This pipeline is currently triggered by the tag push itself, which means this pipeline will race with the "Tests" pipeline, and almost certainly start before the "Tests" pipeline is able to build the image and push it to Docker Hub.

I think we need to change the in-Buildkite configuration so that the tests pipeline triggers this pipeline, on release builds only, _after_ the build step passes. Does that make sense? @philip-stoev can probably translate, if not!

Reply:
That makes sense, I wasn't involved in the release qualification yet, so my assumption was that this is a manually triggered step.

Reply:
The Release Qualification Pipeline is used in the following circumstances:
A. triggered manually
B. Triggered over the weekend
C. Triggered when a git tag is pushed to the main repository.

Lets consider those cases separately:
- For A, we can assume that the image is already available, as the Tests is very likely to have run already against the PR being tested manually
- For B, an unlucky last-minute push to main is going to cause the Release Qualification to fail. This is likely not going to be detected in time and/or restarted, so the entire run will be lost
- For C, we could use @benesch's suggestion

So I suggest using any of the following Buildkite environment variables to distinguish case C from A and B and do not build for C , but build for A and B.

Here are the variables from the "Deploy website" pipeline, which is a triggered one:

```
BUILDKITE_SOURCE="trigger_job"
BUILDKITE_TRIGGERED_FROM_BUILD_ID="018822d7-6881-4afb-a7b9-3820f9635a39"
BUILDKITE_TRIGGERED_FROM_BUILD_NUMBER="55607"
BUILDKITE_TRIGGERED_FROM_BUILD_PIPELINE_SLUG="tests"
```

Here are the variables from a scheduled Nightly run:

```
BUILDKITE_SOURCE="schedule"
```

and here are the variables from a manual run of Nightly on a PR:

```
BUILDKITE_SOURCE="ui"
```

and for the case where the pipeline was triggered because a git tag was pushed:

```
BUILDKITE_SOURCE="webhook"
```

So I think there is enough information in there, just from the `BUILDKITE_SOURCE` variable, to make an informed decision. @def- does that sound reasonable?

1. Use the env variable to disambiguate, and build for `ui` and `schedule`
2. Use the Tests pipeline to trigger the Release Qualification on tags
3. Do not trigger the Release Qualification pipeline directly on tags

Reply:
> Do not trigger the Release Qualification pipeline directly on tags

Done, the rest is done in this change.